### ISSUE TITLE
react-router render(), jsx-no-bind alternative.

### ISSUE THREAD

**Issue by ghost** [yannickcr/eslint-plugin-react#1820]
I opened a Stackoverflow question (https://stackoverflow.com/questions/50777747/react-react-router-render-jsx-no-bind-alternative) for this.

-----------------------

I'm using react-router and I was using this Switch and these routes:

```js
<Switch>
  <Route
    path="/route1"
    render={() => <ComponentOne data={data} history={history} match={match} />}
  />
  
  <Route
    path="/route2"
    render={() => <ComponentTwo data={data} history={history} match={match} />}
  />
</Switch>
```

But after adding https://github.com/yannickcr/eslint-plugin-react is says to me:

```error
[eslint] JSX props should not use arrow functions (react/jsx-no-bind)
```

So I'm wondering if this way below (with no warnings and error) is good and if it is the same as the way before:

```js
<Switch>
  <Route
    path="/route1"
  >
    {() => <ComponentOne data={data} history={history} match={match} />}
  </Route>

  <Route
    path="/route2"
  >
    {() => <ComponentTwo data={data} history={history} match={match} />}
  </Route>
</Switch>
```

**There is a better way to do this?**

**Comment by ljharb** [comment]
You can configure `jsx-no-bind` to allow arrow functions.

However, there's no way to use inline render props that doesn't create a new function on every render (the spirit of the rule). What you'd have to do is:
```jsx
class Foo extends React.Component {
  constructor(...args) {
    super(...args);
    this.renderOne = this.renderOne.bind(this);
    this.renderTwo = this.renderTwo.bind(this);
  }

  renderOne() {
    return <ComponentOne data={data} history={history} match={match} />;
  }

  renderTwo() {
    return <ComponentTwo data={data} history={history} match={match} />;
  }

  render() {
    return (
      <Switch>
        <Route
          path="/route1"
        >
          {this.renderOne}
        </Route>

        <Route
          path="/route2"
        >
          {this.renderTwo}
        </Route>
      </Switch>
    );
  }
}
```

**Comment by johnunclesam** [comment]
Thanks @ljharb, but how can I pass parameters?

`{this.renderOne(data)` is not good...

**Comment by johnunclesam** [comment]
**How much is the gain in terms of performance by implementing this advice of the "constructor"?**

I fixed the parameters using `this.props` in every function anyway...

**Comment by ljharb** [comment]
I've commented there, it's not a good answer.

As for "how much is the gain", there's never a way to quantify performance outside of profiling each specific app.
